### PR TITLE
Recode L050

### PR DESCRIPTION
**Pull request by alanmcruickshank**
This is part of #4031 . Initially I thought we might need to modify L050 because it does overlap somewhat with `LT01` (as it also catches initial trailing whitespace). However - I think that small overlap is ok (and also not new).

This is a less breaking change. Just recode L050 > LT13 and be done 👍

### CODE REVIEW

File: src/sqlfluff/rules/layout/LT13.py [hunk 1]
```diff
@@ -63,7 +63,9 @@ class Rule_L050(BaseRule):
             foo
     """
 
-    groups = ("all",)
+    name = "layout.start-of-file"
```
Comment (WittierDinosaur) [inline src/sqlfluff/rules/layout/LT13.py:66]: shouldn't this be snake case?

Comment (WittierDinosaur) [reply src/sqlfluff/rules/layout/LT13.py:66]: also, possibly a good test to add, as an easy mistake to make when writing a new rule

Comment (alanmcruickshank) [reply src/sqlfluff/rules/layout/LT13.py]: oh - good catch. Yes it should 👍

Comment (alanmcruickshank) [reply src/sqlfluff/rules/layout/LT13.py]: I like the idea of validating rule names too - I'll do that as a new PR though.

**Pull request merged**